src/Geom_Poly.f90

```diff
@@ -2003,6 +2012,35 @@ RECURSIVE FUNCTION isequal_PolygonType(p1,p2) RESULT(bool)
 ENDFUNCTION isequal_PolygonType
 !
 !-------------------------------------------------------------------------------
+!> @brief An overloaded operator routine to assign one polygon type to another.
+!> @param p1 The target polygon type to assign to.
+!> @param p2 The source polygon type to assign from.
+!>
+SUBROUTINE assign_PolygonType(p1,p2)
+  TYPE(PolygonType),INTENT(INOUT) :: p1
+  TYPE(PolygonType),INTENT(IN) :: p2
+
+  CALL p1%clear()
+  IF(p2%isinit) THEN
+    p1%isInit=p2%isInit
+    p1%area=p2%area
+    p1%nVert=p2%nVert
+    p1%nQuadEdge=p2%nQuadEdge
+    p1%centroid=p2%centroid
+    ALLOCATE(p1%vert(p2%nVert))
+    ALLOCATE(p1%edge(2,p2%nVert))
+    p1%vert=p2%vert
+    p1%edge=p2%edge
+    ALLOCATE(p1%quad2edge(p2%nQuadEdge))
+    ALLOCATE(p1%quadEdge(3,p2%nQuadEdge))
+    p1%quad2edge=p2%quad2edge
+    p1%quadEdge=p2%quadEdge
+    p1%nextPoly => p2%nextPoly
+    p1%subRegions => p2%subRegions
```

Comment:
Just want to double check... a shallow copy is the right way to go here?  Or do we want to do a deep copy?  (i.e., allocate `nextPoly` and `subRegions` then assign the `p2` versions to `p1`)

Reply:
Good question.  Let me look through the general pin mesh type and think on how the polygon type is used there.  I think that will dictate whether it should be shallow or deep.

Reply:
Sounds good.  Please add a comment on this routine's header that explicitly states whether it's deep or shallow copy too.

Reply:
I'm going to say that since the `%clear` routine only nullifies, and when the `%subregions` pointer and `%nextpoly` pointer are assigned, they are only allocated, it should be a shallow copy.  Otherwise, if we allocate the pointers when we assign, and we clear the assigned polygon, those allocated pointers will still exist and be lost.

Reply:
Yep, makes sense.  Thanks!